### PR TITLE
NEXT-26976 - Adding two guides for creating own permissions

### PR DESCRIPTION
**Pull request by King-of-Babylon**
Two new guides for creating own permissions via app or plugin for the b2b components (Employee management).

### CODE REVIEW

File: products/extensions/b2b-components/guides/creating-own-permissions-via-plugin.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        ];
+    }
+
+    // This method is called when the PermissionCollectorEvent is triggered
+    public function onAddOwnPermissions(PermissionCollectorEvent $event): void
+    {
+        $collection = $event->getCollection();
+
+        // Here you add your custom permissions to the permission collection
+        $collection->add(new PermissionEventDefinition(self::EMPLOYEE_READ, self::OWN_ENTITY_GROUP, []));
```
Comment (dneustadt) [inline products/extensions/b2b-components/guides/creating-own-permissions-via-plugin.md:39]: Can be simplified:

```php
$collection->addPermission(self::EMPLOYEE_READ, self::OWN_ENTITY_GROUP, []);
```

File: products/extensions/b2b-components/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# B2B components
```
Comment (sushmangupta) [inline products/extensions/b2b-components/README.md:1]: Please add a short description to this page.

File: products/extensions/b2b-components/guides/creating-own-permissions-via-app.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Creating own permissions via plugin
+
+This article explains how to create custom permissions using a app.
```
Comment (sushmangupta) [inline products/extensions/b2b-components/guides/creating-own-permissions-via-app.md:3]:
```suggestion
This article explains how to create custom permissions using an app.
```

File: products/extensions/b2b-components/guides/creating-own-permissions-via-app.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+To add a new permission via an app you need to use the app script hook `b2b-role-permissions` in conjunction with app scripting:
```
Comment (sushmangupta) [inline products/extensions/b2b-components/guides/creating-own-permissions-via-app.md:5]:
```suggestion
To add a new permission via an app, you need to use the app script hook `b2b-role-permissions` in conjunction with app scripting:
```

**Review by sushmangupta — CHANGES_REQUESTED** [review]

File: products/extensions/b2b-components/README.md (hunk 2 above)
Comment (sushmangupta) [inline products/extensions/b2b-components/README.md:1]:
```suggestion
# B2B components

The following articles guide you on creating your own permissions via app or plugin for the b2b components (Employee management).
```

Comment (sushmangupta) [inline products/extensions/b2b-components/README.md:1]: Please provide a short description of what are B2B components .

Comment (King-of-Babylon) [reply products/extensions/b2b-components/README.md:1]: B2B components are all B2B features, which can be activated and deactivated individually.
So idk what I could write about this.... :shrug: 
Would that be enough?:
`Shopware B2B Components are all B2B features, which can be activated and deactivated individually.`
The concept of the feature enabling and disabling is currently under construction, so I can't describe it more detailed :shrug:

**Review by sushmangupta — APPROVED** [review]
Thank you @King-of-Babylon

**Pull request merged**